### ISSUE TITLE
About the effective training batch size and the training time

### ISSUE THREAD

**Issue by Pefect96** [sudraj2002/AWRaCLe#7]
In the paper, the training batch size is 32 (8gpu*4？) However, in the provided script, the batch_size is 4  and the num_gpus is 2.

Which training configuration should I choose?

Besides, How long will the training last?

**Comment by sudraj2002** [comment]
Hi, the effective training batch size was 32 for which we used 8 gpu * 4 and you can use the same. Training takes about 2 days.

**Comment by Pefect96** [comment]
> Hi, the effective training batch size was 32 for which we used 8 gpu * 4 and you can use the same. Training takes about 2 days.

Thanks for your response. Can the 24G gpu be used? Additionally, does the code support gradient accumulation? Because I don't have enough GPUs for training.

Finally, How many epochs are needed for training? In the paper, it is 100. In option.py, it is written as 120. On line 81 of train.py, it is 150. I'm a bit confused because it seems to determine the way the learning rate decays, so I need to check this.

Thanks

**Comment by sudraj2002** [comment]
Yes 24G is sufficient. I think you can use gradient accumulation by passing `accumulate_grad_batches` in the PyTorch Lightning Trainer defined [here](https://github.com/sudraj2002/AWRaCLe/blob/e43160b6aed6b7495761243fc1fbf198978db594/train.py#L105). For epochs use 100. You can use 100 in Line 81, but it would cause the learning rate to reduce too much near the end of training, which is why we used 150.

**Comment by Pefect96** [comment]
> Yes 24G is sufficient. I think you can use gradient accumulation by passing `accumulate_grad_batches` in the PyTorch Lightning Trainer defined [here](https://github.com/sudraj2002/AWRaCLe/blob/e43160b6aed6b7495761243fc1fbf198978db594/train.py#L105). For epochs use 100. You can use 100 in Line 81, but it would cause the learning rate to reduce too much near the end of training, which is why we used 150.

Thank you very much for your reply. Could you please tell me at what epoch the performance begins to approach saturation? Do we really need to train for 100 epochs?

**Comment by sudraj2002** [comment]
I cannot exactly remember when it saturates but I think performance should be good around 40 epochs.

**Comment by Pefect96** [comment]
> I cannot exactly remember when it saturates but I think performance should be good around 40 epochs.

OK, Thanks a lot!!
